test/_helper.bash

```diff
@@ -95,6 +95,15 @@ else
     ./../${BIN_UNDER_TEST:-false} "${ARGS}";
   }
 
+  _app_pipe() {
+     local ARGS="${@:-}"
+    if [[ "${BIN_UNDER_TEST}" != "" ]]; then
+      # remove --json flags
+      ARGS=$(echo "${ARGS}" | sed -E 's,--json,,g')
+    fi
+    ./../${BIN_UNDER_TEST:-false} - <"${ARGS}";
```

Comment:
Minor nit: this isn't a pipe : ] Might want to rename to `_app_redirect`, change to pipe, or test both (I don't believe there's any difference when redirecting STDIN but I'm not sure)

Also extra prefixed space on line 99